frontend/src/bundles/friends/friends-api.ts

```diff
@@ -0,0 +1,119 @@
+import { FriendsApiPath } from '~/bundles/friends/enums/enums.js';
+import { type Http } from '~/framework/http/http.js';
+import { BaseHttpApi } from '~/framework/http-api/http-api.js';
+import { type Storage } from '~/framework/storage/storage.js';
+
+import {
+    type FriendRequestDto,
+    type FriendResponseDto,
+    type PaginationParameters,
+} from './types/types.js';
+
+type Constructor = {
+    baseUrl: string;
+    http: Http;
+    storage: Storage;
+};
+
+class FriendsApi extends BaseHttpApi {
+    public constructor({ baseUrl, http, storage }: Constructor) {
+        super({ path: ApiPath.FRIENDS, baseUrl, http, storage });
+    }
+
+    public async getNotFollowed(payload: PaginationParameters): Promise<{
+        users: FriendResponseDto[];
+        query: PaginationParameters;
+    }> {
+        const response = await this.load(
+            this.getFullEndpoint(
+                `${FriendsApiPath.ROOT}?page=${payload.page}&limit=${payload.limit}`,
+                {},
+            ),
+            {
+                method: 'GET',
+                contentType: ContentType.JSON,
+                hasAuth: true,
+            },
+        );
+
+        return await response.json<{
+            users: FriendResponseDto[];
+            query: PaginationParameters;
+        }>();
+    }
+
+    public async getFollowings(payload: PaginationParameters): Promise<{
+        users: FriendResponseDto[];
+        query: PaginationParameters;
+    }> {
+        const response = await this.load(
+            this.getFullEndpoint(
+                `${FriendsApiPath.FOLLOWINGS}?page=${payload.page}&limit=${payload.limit}`,
+                {},
+            ),
+            {
+                method: 'GET',
+                contentType: ContentType.JSON,
+                hasAuth: true,
+            },
+        );
+
+        return await response.json<{
+            users: FriendResponseDto[];
+            query: PaginationParameters;
+        }>();
+    }
+
+    public async addFollowing(
+        payload: FriendRequestDto,
+    ): Promise<FriendResponseDto> {
+        const response = await this.load(
+            this.getFullEndpoint(FriendsApiPath.FOLLOWINGS, {}),
+            {
+                method: 'POST',
+                contentType: ContentType.JSON,
+                payload: JSON.stringify(payload),
+                hasAuth: true,
+            },
+        );
+        return await response.json<FriendResponseDto>();
+    }
+
+    public async removeFollowing(payload: FriendRequestDto): Promise<number> {
+        const response = await this.load(
+            this.getFullEndpoint(FriendsApiPath.FOLLOWINGS, {}),
+            {
+                method: 'DELETE',
+                contentType: ContentType.JSON,
+                payload: JSON.stringify(payload),
+                hasAuth: true,
+            },
+        );
+        return await response.json<number>();
+    }
+
+    public async loadMore(payload: PaginationParameters): Promise<{
```

Comment:
why do we need this additional method?

Reply:
@dimapopovych I need this to implement functionality for the Load More button

Reply:
But for me it seems unnecessary to have this additional method in service. Why we cannot use getFollowings/getNotFollowed methods? I think we could just create separate action

Reply:
@dimapopovych  in loadMore I load the next 10 users each time. And I add them to the end of the state.

In essence, these are the same getFollowings/getNotFollowed methods, but in them I simply store the state (without the ability to add more).

Then without loadMore, when switching to another tab, I will need to clear the state (then I need an additional method for this)

Maybe I don't fully understand how exactly you think it should be implemented.

Reply:
@VasylynaKolodiy 
I think that have Load More action that depending on active tab will call either getFollowings or getNotFollowed methods from service and append array in state will suffice.
If cannot refactor this quickly can be considered as non blocker too